jlm/llvm/opt/alias-analyses/AgnosticMemoryNodeProvider.cpp

```diff
@@ -69,13 +69,15 @@ class AgnosticMemoryNodeProvisioning final : public MemoryNodeProvisioning
   GetOutputNodes(const rvsdg::output & output) const override
   {
     JLM_ASSERT(is<PointerType>(output.type()));
-    auto & registerNode = PointsToGraph_.GetRegisterNode(output);
 
     util::HashSet<const PointsToGraph::MemoryNode *> memoryNodes;
-    for (auto & memoryNode : registerNode.Targets())
-      memoryNodes.Insert(&memoryNode);
+    if (GetOutputNodesFromRegisterNode(output, memoryNodes))
+      return memoryNodes;
+
+    if (GetOutputNodesFromRegisterSetNode(output, memoryNodes))
+      return memoryNodes;
 
-    return memoryNodes;
+    throw util::error("Cannot find register in points-to graph.");
```

Comment:
This should only be temporary as long as we still have both kind of nodes, i.e., RegisterNodes and RegisterSetNodes, around. Once the RegisterNode class disappeared, this will also disappear again.